### PR TITLE
[Compute] Fix a bug of vm create if endpoint_vm_image_alias_doc is not set in cloud profile

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

https://github.com/Azure/azure-cli/issues/12902
If `endpoint_vm_image_alias_doc` is not set in cloud profile, then cannot create a vm from a custom image, and cannot failover to local copy of image alias. This pr change error to warning to fix this problem.

**Testing Guide**  

```
az cloud register -n NewAzureCloud --endpoint-resource-manager "https://management.azure.com/" --endpoint-management "https://management.core.windows.net/"
az cloud show -n NewAzureCloud -o json
az vm create -g -n --image
```

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_image.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def test_read_images_from_alias_doc(self, mock_urlopen):
         self.assertEqual(parts[2], ubuntu_image['sku'])
         self.assertEqual(parts[3], ubuntu_image['version'])
 
+    @unittest.skip('It will not throw an exception now. Instead, it will failover to a local copy.')
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_image.py:59]: Why not change to test the new behavior ? And do we need to get it back in the future? If not, maybe we can just remove it.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_image.py:59]: OK. I'll try to add one.

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_actions.py [hunk 2]
```diff
@@ -89,25 +89,27 @@ def load_images_from_aliases_doc(cli_ctx, publisher=None, offer=None, sku=None):
...
-        if response.status_code == 200:
-            dic = json.loads(response.content.decode())
-        else:
-            logger.warning("Failed to retrieve image alias doc '%s'. Error: '%s'. Use local copy instead.",
-                           target_url, response)
-            dic = json.loads(alias_json)
-    except requests.exceptions.ConnectionError:
-        logger.warning("Failed to retrieve image alias doc '%s'. Error: 'ConnectionError'. Use local copy instead.",
-                       target_url)
+        logger.warning("'endpoint_vm_image_alias_doc' isn't configured. Please invoke 'az cloud update' to configure "
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_actions.py:96]: curious why it's warning now, prevoiusly is exception

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_actions.py:96]: 2 reasons:
1. A local copy is introduced. It should not halt. It should read local copy when online image alias is not configured or not available.
2. Even if image alias endpoint is not configured, it should not block creating vm from a custom image.

Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/vm/_actions.py:96]: ok, so what's image alias endpoint used for?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_actions.py:96]: It is an URL where the image alias file locates. When using `vm create --image centos`, centos is an alias, it needs to be transformed to a standard form, which consists of publisher, offer, sku, version.

**Comment by yonzhan** [comment]
add to S169

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_image.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_image.py]: Added

**Pull request merged**